Dockerfile

```diff
@@ -1,12 +1,16 @@
 ####################################################################
-FROM ubuntu:22.04 as builder 
+FROM ubuntu:23.04 as builder
```

Comment:
is this needed?

Reply:
yep, arm builds fail on older versions of ubuntu because of problems with dependencies i mentioned that is why im using this in CI as well

Reply:
though now that i fixed everything, its possible i tricked myself when i was wrestling with `werror`

Reply:
Right, i was only thinking about our published image (and we can’t publish linux arm images through GitHub). It’s fine though of course, not a big deal

Reply:
why cant we? so long as the builder installs the right stuff it can definitely build arm images. i have done it myself locally on my amd64 machine. you just need to install qemu and run a few commands... do we not have permissions sufficient enough to do that on the github actions maybe?

Reply:
Sorry, I should take more time to answer.. disregard that, of course we can. How long did the cross compilation take?

Reply:
Relatively to compiling the amd image

Reply:
oh it was quite a bit long on my local machine. amd was done in 10 minutes while the arm one took maybe double that? since im making the virtual env changes i'll get to test it again shortly